tools/online_editor/src/index.ts

```diff
@@ -351,8 +345,12 @@ import slint_init, * as slint from "@preview/slint_wasm_interpreter.js";
     }
 
     if (component !== undefined) {
-      const instance = component.create(canvas_id);
-      instance.show();
+      if (current_preview_instance === null) {
+        current_preview_instance = component.create("rendering_canvas");
+        current_preview_instance.show();
```

Comment:
Can we do `run` here and get rid of the `if` staring in line `354`?

Reply:
I'd rather remove `run()` altogether, because it has different semantics that `run()` usually. In all our other apis `run()` is basically `show(); run_event_loop(); hide(); return` but here it's not - it "returns" by throwing an exception and never calls hide. I wanted to clean that up in a separate patch.